streaming-payments.md

```diff
@@ -0,0 +1,375 @@
+          console.error(err.message)
+        })
+      } else {
+        console.log('no match!', { secret, fulfillment, condition, transfer })
+      }
+    }
+  })
+
+  http.createServer(function (req, res) {
+    const secret = crypto.randomBytes(32)
+    const user = base64(crypto.randomBytes(8))
+    users[user] = { secret, res }
+    console.log('user! writing head', user)
+    res.writeHead(200, {
+      'Pay': [ 1, plugin.getAccount() + '.' + user, base64(secret) ].join(' ')
+    })
+    // Flush the headers in a first TCP packet:
+    res.socket.write(res._header)
+    res._headerSent = true
+  }).listen(8000)
+})
+```
+
+Starting with the last part, `http.createServer`, you can see the flow of the http server
+is a bit simpler; when a request comes in it sends headers, and
+then the body will be sent letter-by-letter, as payments come in. The `Pay` header contains 3 parts:
+
+* amount (in this case, the price of one letter, in XRP)
+* user-specific destination address
+* a Base64-encoded conditionSeed
+
+Note how we are appending `'.' + user` to the shop's Interledger address! This is a special feature of Interledger
+addresses, they can be subnetted endlessly, just add another `.` at the end to convert an account address
+to a ledger prefix, and then add new sub-accounts after that. In this case, we want to know which user is paying
+for letters, so by telling each user a different Interledger sub-address, we can neatly keep our users apart.
+
+When a transfer comes in, the server opens the [ILP packet](https://github.com/interledgerjs/ilp-packet).
+In there is a destination account and a destination amount, which will usually (if all went well) be equal
+to the amount of the transfer. In later tutorials, we will see how one ILP payment can be chained together
+in multiple transfers, but for now, we will only use single-transfer payments.
+
+Whereas in the previous tutorial, the fulfillment was generated randomly, here it's generated deterministically from
+the combination of the shop's interledger address, the user's userId, this payment's paymentId, and the
+conditionSeed (as a buffer) that was sent to this user base64-encoded in the `Pay` header, for example:
+
+```js
+amount = '1'
+shopAddress = 'test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW'
+userId = 'UUxOFrawp0A'
+conditionSeed = Buffer.from('Agli74z_HjSNuTE1rTIr7QCkzWdIA3QdKoPMYHhw1I4', 'base64')
+xPayHeader = [ amount, shopAddress + '.' + userId, base64(conditionSeed) ]
+paymentId = '1'
+paymentPacketContents = {
+  amount: xPayHeader[0],
+  account: xPayHeader[1] + '.' + paymentId,
+  data: ''
+}
+fulfillment = hmac(conditionSeed, paymentPacketContents.account)
+condition = sha256(fulfillment)
+```
+
+To run this new version of the shop, type this into your terminal:
+
+```sh
+npm install ilp-packet
+node ./shop2.js
+```
+
+The ILP paacket is not very useful in a one-transfer payment, but in future tutorials, we see how connectors can forward an
```

Comment:
> paacket

Is that the Dutch version of the word? 😉

Reply:
haha, no, the Dutch version is "pakket". :)